prost-build/build.rs

```diff
@@ -1,44 +1,54 @@
-extern crate curl;
-extern crate tempdir;
-extern crate zip;
+extern crate fs_extra;
+
+use fs_extra::dir;
 
 use std::env;
 use std::fs;
-use std::io::{
-    self,
-    Cursor,
-};
 use std::path::{
-    Path,
     PathBuf,
 };
 
-use curl::easy::Easy;
-use tempdir::TempDir;
-use zip::ZipArchive;
-
-const VERSION: &'static str = "3.5.1";
-
 fn main() {
     if env_contains_protoc() { return; }
 
-    let out_dir = PathBuf::from(env::var("OUT_DIR").expect("OUT_DIR environment variable is invalid"));
-    let protoc_dir = out_dir.join("protoc");
+    // Recursively copy all the protobuf files to `OUT_DIR` so that they are
+    // available for use at runtime.
 
-    // If the protoc directory doesn't already exist from a previous build, download protoc.
-    if !protoc_dir.exists() {
-        // The unzipping done by download_protoc is not atomic, so to avoid failing halfway through
-        // and leaving trash state, download/unzip to a temporary directory, then rename.
-        let tempdir = TempDir::new_in(&out_dir, "protoc").expect("failed to create temporary directory");
-        download_protoc(tempdir.path());
-        fs::rename(&tempdir.into_path(), &protoc_dir).expect("unable to move temporary directory");
-    }
-    let mut protoc = protoc_dir.join("bin");
-    protoc.push("protoc");
-    protoc.set_extension(env::consts::EXE_EXTENSION);
+    let src_dir = PathBuf::from("../third-party/protobuf");
+    static INCLUDE: &str = "include";
+
+    let protoc_bin_name = match (env::consts::OS, env::consts::ARCH) {
+        ("linux", "x86") => "protoc-linux-x86_32",
+        ("linux", "x86_64") => "protoc-linux-x86_64",
+        ("linux", "aarch64") => "protoc-linux-aarch_64",
+        ("macos", _) => "protoc-osx-x86_64",
```

Comment:
Will an x86_64 binary run on a `macos` 32bit system?  I'm behind the decision to drop 32bit support, but I think it'd be better to have this panic fire in that case, since it should be easier to debug than whatever error is presented when trying to execute 64bit binary on a 32bit system.

Reply:
to be clear, I'm wondering if the match on line 24 should be `("macos", "x86_64")`.